xdsl/transforms/convert_snrt_to_riscv.py

```diff
@@ -14,6 +15,19 @@
 )
 
 
+@dataclass(frozen=True)
+class SnrtConstants:
+    """
+    Constants used when compiling the snitch runtime, depend on the exact snitch
+    architecture target.
+    """
+
+    SNRT_CLUSTER_NUM: int = 4
+    SNRT_CLUSTER_CORE_NUM: int = 9
+    SNRT_BASE_HARTID: int = 0
+    SNRT_CLUSTER_DM_CORE_NUM: int = 1
```

Comment:
this feels weird, shouldn't these be lower_snake_case?

Reply:
Also, I'd recommend removing the default values here, since they're duplicated in the pass. Another alternative would be to move these constants to the module level, and reusing them as default values on the pass. One concern is there might be a new constant added in the future, and it would be nice for the type system to tell us that the pass isn't passing this new value along, instead of silently just using a different value to what's expected.

Reply:
Hmm, I have an idea :bulb: